lib/paraactions/paraactions.ts

```diff
@@ -0,0 +1,790 @@
+// lib/paraactions/paraactions.ts
+// Small DSL parser + executor for declarative ParaCharts actions,
+// with support for braced `when(...) { ... }` blocks and nesting.
+//
+// See previous comments in this file header for full language description.
+
+export type ParaValueKind =
+  | 'string'
+  | 'number'
+  | 'boolean'
+  | 'null'
+  | 'identifier';
+
+export interface ParaValue {
+  valueKind: ParaValueKind;
+  value: string | number | boolean | null;
+}
+
+// Arguments
+export interface ParaPositionalArg extends ParaValue {
+  kind: 'positional';
+}
+
+export interface ParaNamedArg extends ParaValue {
+  kind: 'named';
+  name: string;
+}
+
+export type ParaActionArg = ParaPositionalArg | ParaNamedArg;
+
+// AST for simple actions
+export interface NamedAction {
+  kind: 'namedAction';
+  name: string;
+  args: ParaActionArg[];
+}
+
+// One segment of a chain: methodName(args...)
```

Comment:
We could simplify by merging `NamedAction` and `ChainSegment` so that `ChainAction` contained an array of `NamedAction`s.